sdk/trace/provider.go

```diff
@@ -177,7 +192,7 @@ func (p *TracerProvider) RegisterSpanProcessor(sp SpanProcessor) {
 func (p *TracerProvider) UnregisterSpanProcessor(sp SpanProcessor) {
 	p.mu.Lock()
 	defer p.mu.Unlock()
-	if p.isShutdown {
+	if p.isShutdown.Load() {
```

Comment:
We can also add

```golang
if p.isShutdown.Load() {
  return
```

before aquiring the lock (`p.mu.Lock()`)

Then we will have the "book" implementation of double-checked locking 😉

Reply:
I think here the same problem exits as in `Tracer()`

> // This check happens before the mutex is acquired to avoid deadlocking if Tracer() is called from within Shutdown().

as external code is called:

```
			if err := sp.Shutdown(context.Background()); err != nil {
				otel.Handle(err)
			}
```